### ISSUE TITLE
MP4 Uploads To S3 Failing

### ISSUE THREAD

**Issue by xander-carruth** [aws-amplify/amplify-swift#3121]
### Describe the bug

Hi so I'm using Amplify version 2.15.0 in Swift to attempt to upload a video to S3. My configuration is working fine for photos and text files, however when I attempt to upload an MP4, I get the following error: 
`Unexpected error occurred with message: Response is not an HTTP response`
`Recovery suggestion: This should not happen. There is a possibility that there is a bug if this error persists`
I am wondering if this is actually a bug as I double and triple checked my file paths and my IAM upload permissions. I have tried the upload in the following configurations:
`let options = StorageUploadFileRequest.Options(accessLevel: .guest, contentType: "video/mp4")
 let uploadTask = Amplify.Storage.uploadFile(key: postPath, local: url, options: options)`
And
`let uploadTask = Amplify.Storage.uploadFile(key: postPath, local: url)`

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Set up S3 configuration and IAM roles
2. Store your video you want to upload in a local url or grab the local url from photo assets
3. Give the video the following filepath: "videos/\(cognitioIdentitySub)/999612ef-95bf-407e-be00-6d899c5a7e06cee569fb-21cc-4ad4-8cd4-b8cf4561047e2023-08-03T12:20:25.131Z.mp4"
4. Attempt to upload the video with the following code:
`let uploadTask = Amplify.Storage.uploadFile(key: postPath, local: url)`
```

### Expected behavior

The mp4 video should successfully upload to S3

### Amplify Framework Version

2.15.0

### Amplify Categories

Storage

### Dependency manager

Swift PM

### Swift version

5.7.2

### CLI version

Not using CLI

### Xcode version

14.3.1

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


StorageError: Unexpected error occurred with message: Response is not an HTTP response
Recovery suggestion: This should not happen. There is a possibility that there is a bug if this error persists. Please take a look at https://github.com/aws-amplify/amplify-ios/issues to see if there are any existing issues that match your scenario, and file an issue with the details of the bug if there isn't. Issue encountered at:
file: /Users/xandercarruth/Library/Developer/Xcode/DerivedData/Salonn-bzvuiifqjhrfgvbhtrjvjropsort/SourcePackages/checkouts/amplify-ios/Amplify/Categories/Storage/Error/StorageError.swift
function: recoverySuggestion
line: 104
Caused by:
Error Domain=NSURLErrorDomain Code=-1 "unknown error" UserInfo={NSErrorFailingURLStringKey={my url is here and I am unsure if this is safe to publicly share here}
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

iOS 14.0+

### Device

Multiple

### Specific to simulators

No

### Additional context

Please let me know if any additional information is needed! Thank you!

**Comment by harsh62** [comment]
@xander-carruth Thanks for raising this issue. I will try to investigate and reply on the issue as soon as I have more information.

**Comment by phantumcode** [comment]
Hi @xander-carruth I'm not able to reproduce the issue you're running into so far.  I've been able to successfully upload an mp4 video into S3 using Amplify Storage.

Can you provide additional details to help troubleshoot?
1.  "videos/\(cognitioIdentitySub)/999612ef-95bf-407e-be00-6d899c5a7e06cee569fb-21cc-4ad4-8cd4-b8cf4561047e2023-08-03T12:20:25.131Z.mp4" This is the path you're storing in the video in S3?
2. How large are the video files?
3. Are you uploading as a guest/unathenticated user or authenticated user?
4. Have you tried disabling all IAM permission policies on S3 and allowing all access to to S3 to verify if it is the IAM permission that is the issue?
5. You can enable verbose log by adding the line `Amplify.Logging.logLevel = .verbose` when initializing Amplify.  This may capture more logs to help with the troubleshooting.

**Comment by xander-carruth** [comment]
Hi Tuan, thanks for the quick response!
Everything was correct with the IAM but I was only testing everything with one video file. I didn't really think about the fact that it could be the file because when I was using it with Firebase Storage it was working just fine. I switched to another video file for testing and the uploads are working now. It must have just been something with that specific video file, it was shot on iPhone and only 1.5MB so I wonder what that might have been. I tried to switch the log level to verbose to get more information but no additional information was provided on the error. I think this issue is fine to be closed now since the rest of video uploads are working, unless you would like to look at the video file and see why that was happening.
